### PR TITLE
doc: update ivshmem feature HLD

### PR DESCRIPTION
**Pull request by yonghuah**
With ivshmem doorbell feature enabled, HDL of it shall be updated.

Signed-off-by: Yonghua Huang <yonghua.huang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 1]
```diff
@@ -47,14 +47,20 @@ ivshmem dm:
 ivshmem server:
    A daemon for inter-VM notification capability that will work with **ivshmem
    dm**. This is currently **not implemented**, so the inter-VM communication
-   doesn't support a notification mechanism.
+   doesn't support a notification mechanism for dm-land ivshmem devices.
```
Comment (dbkinder) [inline doc/developer-guides/hld/ivshmem-hld.rst:50]: I'd suggest rewriting this as:
```
A daemon for inter-VM notification capability that will work with **ivshmem
dm**.  This is currently not implemented for **dm-land** ivshmem devices.

For **hv-land** ivhsmem devices, a notification mechanism (doorbell) is enabled.
The VM triggers the notification by writing the target peer ID (VM ID) and 
vector index to the doorbell register.  The Hypervisor finds the target
ivshmem device by this peer ID and injects a virtual interrupt to the target VM. 
```

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 2]
```diff
@@ -47,14 +47,20 @@ ivshmem dm:
...
+   to doorbell register, Hypervisor will find target ivshmem device by this peer ID
+   and inject virtual interrupt to the target VM.
 
 Ivshmem Device Introduction
 ***************************
 
 The ``ivshmem`` device is a virtual standard PCI device consisting of
-two Base Address Registers (BARs): BAR0 is used for emulating interrupt
-related registers, and BAR2 is used for exposing shared memory region. The ``ivshmem`` device doesn't support any extra capabilities.
+three Base Address Registers (BARs): BAR0 is used for emulating interrupt
```
Comment (dbkinder) [inline doc/developer-guides/hld/ivshmem-hld.rst:61]: Suggested rewrite:
```
The ``ivshmem`` device is a virtual standard PCI device consisting of
three Base Address Registers (BARs): 

* BAR0 is used for emulating interrupt related registers, 
* BAR1 is used for emulating MSIX entry table, and
* BAR2 is used for exposing a shared memory region.

The ``ivshmem`` device supports no extra capabilities.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for the doc update.

**Comment by yonghuah** [comment]
@dbkinder  have updated this PR, please review ,thanks.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Review by gvancuts — COMMENTED** [review]
The text looks good. I'm not sure I fully understand the diagram at the beginning of the document as it shows an `ivshmem server` running in the Service VM that interacts only with acrn-dm` instances. Are there details missing to describe this new feature?

**Comment by yonghuah** [comment]
@gvancuts  thanks for comments,  I refined  some desc. pls review, thanks.

**Comment by gvancuts** [comment]
Thanks @yonghuah . If you have the ability to add a box showing this `ivshmem server` module in the hypervisor box, that would be even better, but not a blocker for this PR. Thanks! 👍

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Review by dbkinder — APPROVED** [review]
Thanks for fixing the other areas in the doc that talked about this.  LGTM now.

**Pull request merged**